### PR TITLE
Add DinaHosting

### PR DESCRIPTION
**Pull request by reetp**
I hope this fixes:

https://github.com/ddclient/ddclient/issues/203

### CODE REVIEW

File: ddclient.in [hunk 1]
```diff
@@ -710,6 +719,15 @@ my %services = (
...
         },
     },
+    'dinahosting' => {
+     'updateable' => undef,
+     'update'     => \&nic_dinahosting_update,
+     'examples'   => \&nic_dinahosting_examples,
+     'variables'  => merge(
+                 $variables{'dinahosting-service-common-defaults'},
+                 $variables{'service-common-defaults'},
+                  ),
```
Comment (rhansen) [inline ddclient.in:729]: Move the `dinahosting-service-common-defaults` variables to here, like this:

```suggestion
        'variables' => {
            %{$variables{'service-common-defaults'}},
            'max-interval'              => setv(T_DELAY,  0, 0, 1, interval('25d'),    0),
            'min-error-interval'        => setv(T_DELAY,  0, 0, 1, interval('8m'),     0),
            'min-interval'              => setv(T_DELAY,  0, 0, 1, interval('30s'),    0),
            'script'                    => setv(T_STRING, 0, 1, 1, '/special/api.php', undef),
            'server'                    => setv(T_FQDNP,  1, 0, 1, 'dinahosting.com',  undef),
            'warned-min-error-interval' => setv(T_ANY,    0, 1, 0, 0,                  undef),
            'warned-min-interval'       => setv(T_ANY,    0, 1, 0, 0,                  undef),
        },
```

File: ddclient.in [hunk 2]
```diff
@@ -2849,6 +2867,85 @@ sub nic_dyndns2_update {
...
+
+               $config{$h}{'status'} = 'failed';
+               failed("updating %s: %s - %s", $h, $statuscode, $statusmessage);
+          }
+     }
+}
+######################################################################
+## nic_dinahosting_examples
+######################################################################
+sub nic_dinahosting_examples {
```
Comment (rhansen) [inline ddclient.in:2926]: Move this function above `nic_dinahosting_update` (to match what other protocols do).

File: ddclient.in [hunk 3]
```diff
@@ -2849,6 +2867,85 @@ sub nic_dyndns2_update {
...
+    return <<EoEXAMPLE;
+o 'dinahosting'
+
+The 'dinahosting' protocol is used by DNS services offered by www.dinahosting.com.
```
Comment (rhansen) [inline ddclient.in:2930]: Add a link to https://en.dinahosting.com/api

File: ddclient.in [hunk 4]
```diff
@@ -2849,6 +2867,85 @@ sub nic_dyndns2_update {
...
+
+Configuration variables applicable to the 'dinahosting' protocol are:
+  protocol=dinahosting         ##
+  server=fqdn.of.service       ## defaults to dinahosting.com
+  login=fully.qualified.host   ## login name and password registered with the service
```
Comment (rhansen) [inline ddclient.in:2935]: Should this be `server=myusername` or does dinahosting use the hostname as the username?

File: ddclient.in [hunk 5]
```diff
@@ -2849,6 +2867,85 @@ sub nic_dyndns2_update {
...
+          # alexandregz: dinahosting uses subdomain into another param
+          my ($subdomain, $domain) = split(/\./, $h, 2);
+
+          # Set the URL that we're going to update
+          my $url = "https://$config{$h}{'server'}$config{$h}{'script'}";
+          $url  .= "?AUTH_USER=$config{$h}{'login'}";
+          $url  .= "&AUTH_PWD=$config{$h}{'password'}";
+          $url  .= "&hostname=$subdomain";
+          $url  .= "&domain=$domain";
+          $url  .= "&command=Domain_Zone_UpdateTypeA";  # new zones change to Domain_Zone_AddTypeA
```
Comment (rhansen) [inline ddclient.in:2891]: This should be `Domain_Zone_UpdateTypeAAAA` if `$ip` is an IPv6 address:
```suggestion
          $url  .= "&command=Domain_Zone_UpdateType" . is_ipv6($ip) ? "AAAA" : "A";
```

File: ddclient.in (hunk 4 above)
Comment (reetp) [reply ddclient.in:2935]: `server=www.dinahosting.com`

**Comment by reetp** [comment]
> Thank you for your contribution!
> 

NP! Most of it was down to another so I can't claim all the credit here!

> There are some small nits here and there; would you be comfortable with me making those changes on your branch? Or would you prefer me to tell you about them so that you can make them?

Please, go for it - you know it waaaaay better than me and you are far more competent than me!

It will be quicker and neater if you do it!

File: ddclient.in (hunk 4 above)
Comment (rhansen) [reply ddclient.in:2935]: Oops, I mean: Should this be `login=myusername` or does dinahosting use the hostname as the username?

**Comment by reetp** [comment]
Many thanks.

Will do. May take me a couple of days.

**Pull request merged**